### ISSUE TITLE
tensorflow/stream_executor/cuda/cuda_dnn.cc:82] The primary convolution algorithm failed memory allocation, while a secondary algorithm is not provided

### ISSUE THREAD

**Issue by MN5930** [tensorflow/tensorflow#25757]
I am running Keras with TF backend on venv Python 3.7.2.  

**System information**
** pip list: 
absl-py              0.6.1      
astor                0.7.1      
backcall             0.1.0      
biwrap               0.1.6      
bleach               3.0.2      
cloudpickle          0.6.1      
cycler               0.10.0     
dask                 1.0.0      
decorator            4.3.0      
defusedxml           0.5.0      
entrypoints          0.2.3      
gast                 0.2.0      
grpcio               1.17.0     
h5py                 2.8.0      
imageio              2.4.1      
imgaug               0.2.6      
ipykernel            5.1.0      
ipython              7.2.0      
ipython-genutils     0.2.0      
ipywidgets           7.4.2      
jedi                 0.13.1     
Jinja2               2.10       
jsonschema           2.6.0      
jupyter              1.0.0      
jupyter-client       5.2.4      
jupyter-console      6.0.0      
jupyter-core         4.4.0      
Keras                2.1.3      
Keras-Applications   1.0.6      
Keras-Preprocessing  1.0.5      
kiwisolver           1.0.1      
Markdown             3.0.1      
MarkupSafe           1.1.0      
matplotlib           3.0.2      
mistune              0.8.4      
mock                 2.0.0      
nbconvert            5.4.0      
nbformat             4.4.0      
networkx             2.2        
notebook             5.7.2      
numpy                1.15.4     
opencv-python        3.4.4.19   
pandas               0.24.1     
pandocfilters        1.4.2      
parso                0.3.1      
pbr                  5.1.1      
pexpect              4.6.0      
pickleshare          0.7.5      
Pillow               5.3.0      
pip                  18.1       
prometheus-client    0.5.0      
prompt-toolkit       2.0.7      
protobuf             3.6.1      
ptyprocess           0.6.0      
pydicom              1.2.1      
Pygments             2.3.0      
pyparsing            2.3.0      
python-dateutil      2.7.5      
pytz                 2018.9     
PyWavelets           1.0.1      
PyYAML               3.13       
pyzmq                17.1.2     
qtconsole            4.4.3      
scikit-image         0.14.1     
scipy                1.1.0      
Send2Trash           1.5.0      
setuptools           39.0.1     
Shapely              1.6.4.post2
six                  1.12.0     
tensorboard          1.12.0     
tensorflow           1.12.0rc0  
tensorflow-estimator 1.10.12    
tensorflow-plot      0.3.0.dev0 
termcolor            1.1.0      
terminado            0.8.1      
testpath             0.4.2      
toolz                0.9.0      
tornado              5.1.1      
traitlets            4.3.2      
wcwidth              0.1.7      
webencodings         0.5.1      
Werkzeug             0.14.1     
wheel                0.32.3     
widgetsnbextension   3.4.2 

** python3 --version
Python 3.7.2

** lsb_release -a
Description:	Ubuntu 16.04.5 LTS
Release:	16.04
Codename:	xenial

** lspci | grep VGA
01:00.0 VGA compatible controller: NVIDIA Corporation Device 1b02 (rev a1)
02:00.0 VGA compatible controller: NVIDIA Corporation Device 1b02 (rev a1)
 
** VGAs detected by TF
2019-02-14 13:56:21.125443: I tensorflow/compiler/xla/service/service.cc:158]   StreamExecutor device (0): TITAN Xp, Compute Capability 6.1
2019-02-14 13:56:21.125447: I tensorflow/compiler/xla/service/service.cc:158]   StreamExecutor device (1): TITAN Xp, Compute Capability 6.1
2019-02-14 13:56:21.125746: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1431] Found device 0 with properties: 
name: TITAN Xp major: 6 minor: 1 memoryClockRate(GHz): 1.582
pciBusID: 0000:01:00.0
totalMemory: 11.91GiB freeMemory: 11.15GiB
2019-02-14 13:56:21.125898: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1431] Found device 1 with properties: 
name: TITAN Xp major: 6 minor: 1 memoryClockRate(GHz): 1.582
pciBusID: 0000:02:00.0
totalMemory: 11.91GiB freeMemory: 11.75GiB

** python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"
b'v1.12.0-rc0-4136-g840be30' 1.12.0-rc0

** cat /usr/local/cuda-10.0/include/cudnn.h | grep CUDNN_MAJOR -A 2
#define CUDNN_MAJOR 7
#define CUDNN_MINOR 4
#define CUDNN_PATCHLEVEL 1
--
#define CUDNN_VERSION (CUDNN_MAJOR * 1000 + CUDNN_MINOR * 100 + CUDNN_PATCHLEVEL)
#include "driver_types.h"

**  nvcc --version
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2018 NVIDIA Corporation
Built on Sat_Aug_25_21:08:01_CDT_2018
Cuda compilation tools, release 10.0, V10.0.130

After couple of iterations within the first epoch of training I get the error "E tensorflow/stream_executor/cuda/cuda_dnn.cc:82] The primary convolution algorithm failed memory allocation, while a secondary algorithm is not provided" then my loss function becomes "nan".
